2020-06-01-DANCE-1-eMolecules-t142-selected/02_create_torsiondrive_dataset.py

```diff
@@ -0,0 +1,149 @@
+"""Runs the torsion drive calculations."""
+import gzip
+import json
+
+import tqdm
+
+import qcelemental as qcel
+import qcportal as ptl
+from qcelemental.models import Molecule
+
+COLLECTION_NAME = "DANCE 1 eMolecules t142 selected"
+INPUT_FILE = "optimization_inputs.json.gz"
+UPDATE = False
+
+
+def read_optimization_inputs(input_json_gz):
+    """ Read data generated by 01_generate.py
+    Returns
+    -------
+    optimization_inputs: dict
+        Dictionary for selected torsions, has this structure:
+        [
+            {
+                'cmiles_identifiers': {
+                    "canonical_smiles": ...,
+                    ..
+                },
+                'atom_indices': [
+                    [
+                        1,
+                        2,
+                        3,
+                        4
+                    ]
+                ],
+                'initial_molecules': [
+                    Molecule1a,
+                    Molecule1b,
+                    ..
+                ]
+            },
+            ..
+        ]
+    """
+    with gzip.open(input_json_gz, 'r') as f:
+        optimization_inputs = json.load(f)
+    return optimization_inputs
+
+
+def main():
+    """Loads the input and runs all the calculations."""
+    print("Reading optimization_inputs...")
+    optimization_inputs = read_optimization_inputs(INPUT_FILE)
+    print(f"Found {len(optimization_inputs)} torsions")
+
+    print("Initializing dataset...")
+    #  client = ptl.FractalClient("localhost:7777", verify=False)
```

Comment:
Is this the correct way to create the client?

Reply:
Yup!